src/connection_manager.c

```diff
@@ -837,6 +877,23 @@ void qd_connection_manager_free(qd_connection_manager_t *cm)
         connector = DEQ_HEAD(cm->connectors);
     }
 
+    connector = DEQ_HEAD(cm->data_connectors);
```

Comment:
nit: if you use a local list and append cm->connectors and cm->data_connectors can this code duplication be removed?